activemodel/lib/active_model/errors.rb

```diff
@@ -93,6 +93,18 @@ def copy!(other) # :nodoc:
       @details  = other.details.dup
     end
 
+    # Merges the errors from <tt>other</tt>.
+    #
+    # other - The ActiveModel::Errors instance.
+    #
+    # Examples
+    #
+    #   person.errors.merge!(other)
+    def merge!(other)
+      @messages.merge!(other.messages) { |_, hash1, hash2| hash1 + hash2 }
```

Comment:
Are the variable inside the block hashes or arrays? I think they are arrays, so maybe we should rename them?